### PR TITLE
Add font increase and decrease to toolbar

### PR DESCRIPTION
**Pull request by mklewandowski**
Add toolbar buttons to increase the overall font size of text in the editor component.  This implementation uses the slate plugin model where font adjustment commands are used by the tooolbar to adjust the font sizes.  Font size is stored in the slate editor data field and retrieved by our React component to render things accordingly.

### CODE REVIEW

File: src/plugins/font-size-plugin.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    increaseFontSize: function (editor: Editor) {
+      const editorData: Data = editor.value.data;
+      const currentFontSize = editorData.has("fontSize") ? editorData.get("fontSize") : kInitialSize;
+      const newFontSize = Math.min(currentFontSize + kFontSizeDelta, kFontSizeMaximum);
+      editor.setData(editorData.set("fontSize", newFontSize));
+      return editor;
+    },
+    decreaseFontSize: function (editor: Editor) {
+      const editorData: Data = editor.value.data;
+      const currentFontSize = editorData.has("fontSize") ? editorData.get("fontSize") : kInitialSize;
```
Comment (kswenson) [inline src/plugins/font-size-plugin.tsx:20]:
```suggestion
      const currentFontSize = editorData.get("fontSize") || kInitialSize;
```

File: src/assets/icon-font-decrease.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+// FontAwesome icon: https://www.iconfinder.com/icons/1608420/minus_search_icon
+// initial JSX conversion: https://react-svgr.com/playground/
+export default function IconFontDecrease(props: IconProps) {
+  return (
+    <svg height={1792} viewBox="0 0 1792 1792" width={1792} {...props}>
+      <path d="M1088 800v64q0 13-9.5 22.5T1056 896H480q-13 0-22.5-9.5T448 864v-64q0-13 9.5-22.5T480 768h576q13 0 22.5 9.5t9.5 22.5zm128 32q0-185-131.5-316.5T768 384 451.5 515.5 320 832t131.5 316.5T768 1280t316.5-131.5T1216 832zm512 832q0 53-37.5 90.5T1600 1792q-54 0-90-38l-343-342q-179 124-399 124-143 0-273.5-55.5t-225-150-150-225T64 832t55.5-273.5 150-225 225-150T768 128t273.5 55.5 225 150 150 225T1472 832q0 220-124 399l343 343q37 37 37 90z" />
+    </svg>
+  );
+}
```
Comment (kswenson) [inline src/assets/icon-font-decrease.tsx:12]: Add newline so we don't get these symbols in GitHub. We should probably enable that lint setting.

File: src/assets/icon-font-increase.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+// FontAwesome icon: https://www.iconfinder.com/icons/1608420/minus_search_icon
+// initial JSX conversion: https://react-svgr.com/playground/
+export default function IconFontIncrease(props: IconProps) {
+  return (
+    <svg height={1792} viewBox="0 0 1792 1792" width={1792} {...props}>
+      <path d="M1088 800v64q0 13-9.5 22.5T1056 896H832v224q0 13-9.5 22.5T800 1152h-64q-13 0-22.5-9.5T704 1120V896H480q-13 0-22.5-9.5T448 864v-64q0-13 9.5-22.5T480 768h224V544q0-13 9.5-22.5T736 512h64q13 0 22.5 9.5T832 544v224h224q13 0 22.5 9.5t9.5 22.5zm128 32q0-185-131.5-316.5T768 384 451.5 515.5 320 832t131.5 316.5T768 1280t316.5-131.5T1216 832zm512 832q0 53-37.5 90.5T1600 1792q-54 0-90-38l-343-342q-179 124-399 124-143 0-273.5-55.5t-225-150-150-225T64 832t55.5-273.5 150-225 225-150T768 128t273.5 55.5 225 150 150 225T1472 832q0 220-124 399l343 343q37 37 37 90z" />
+    </svg>
+  );
+}
```
Comment (kswenson) [inline src/assets/icon-font-increase.tsx:12]: Add newline so we don't get these symbols in GitHub. We should probably enable that lint setting.

File: src/assets/icon-font-decrease.tsx (hunk 2 above)
Comment (mklewandowski) [reply src/assets/icon-font-decrease.tsx]: Fixed.

File: src/assets/icon-font-increase.tsx (hunk 3 above)
Comment (mklewandowski) [reply src/assets/icon-font-increase.tsx]: Fixed.

File: src/plugins/font-size-plugin.tsx (hunk 1 above)
Comment (mklewandowski) [reply src/plugins/font-size-plugin.tsx]: Fixed.

File: src/plugins/font-size-plugin.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+const kFontSizeMinimum = .2;
+const kFontSizeMaximum = 4;
+const kFontSizeDelta = .1;
+export const kInitialSize = 1;
+
+export const fontSizePlugin: Plugin = {
+  commands: {
+    increaseFontSize: function (editor: Editor) {
+      const editorData: Data = editor.value.data;
+      const currentFontSize = editorData.has("fontSize") ? editorData.get("fontSize") : kInitialSize;
```
[the remark this answers is not in the document]
Comment (mklewandowski) [reply src/plugins/font-size-plugin.tsx]: Fixed.

File: src/slate-editor/slate-editor.tsx [hunk 5]
```diff
@@ -69,6 +71,9 @@ const SlateEditor: React.FC<IProps> = (props: IProps) => {
                   : props.value || kEmptyEditorValue;
   const [prevValue, setPrevValue] = useState<EditorValue>(value);
 
+  const fontSize = value.data.has("fontSize") ? value.data.get("fontSize") : kInitialSize;
+  const style = {fontSize: `${fontSize}em`};
```
[the remark this answers is not in the document]
Comment (mklewandowski) [reply src/slate-editor/slate-editor.tsx]: Agreed.  Changed to not pass prop when we don't have a fontSize in the data.

File: src/plugins/font-size-plugin.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import { Editor, Plugin } from "slate-react";
+import { Data } from "slate";
+
+const kFontSizeMinimum = .2;
+const kFontSizeMaximum = 4;
+const kFontSizeDelta = .1;
+export const kInitialSize = 1;
```
[the remark this answers is not in the document]
Comment (mklewandowski) [reply src/plugins/font-size-plugin.tsx]: Fixed.

**Pull request merged**